mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -726,6 +887,76 @@ def set_tag(self, run_id, tag):
             self._check_run_is_active(run)
             session.merge(SqlTag(run_uuid=run_id, key=tag.key, value=tag.value))
 
+    def _set_tags(self, run_id, tags, attempt=None):
+        """
+        Set multiple tags on a run
+
+        :param run_id: String ID of the run
+        :param tags: List of RunTag instances to log
+        """
+        if not tags:
+            return
+
+        # use SessionManager instead of ManagedSessionManager to set
+        # isolation level for individual transaction.
+        # Here, we open a connection with SERIALIZABLE isolation level.
+        # it will be released when session.commit is called.
+        session = self.SessionMaker()
```

Comment:
We don't need a serializable transaction for log_params and set_tags because we're only writing to a single table and we aren't performing any gap locking (e.g. `select for update`). Can we use `ManagedSessionMaker` instead?

Reply:
Done